UMLEditor/UMLEditor/Classes/Class.cs

```diff
@@ -94,6 +94,22 @@ public void AddMethod(string returnType, string name)
         
     }
 
+    /// <summary>
+    /// Adds method to class.
+    /// Pre-condition: name of method is valid
+    /// </summary>
+    /// <param name="returnType">Valid returnType of new method</param>
+    /// <param name="name">Valid name of new method</param>
+    /// <param name="paramList">List of parameters</param>
+    public void AddMethod(string returnType, string name, List<NameTypeObject> paramList)
+    {
+        if (MethodExists(name))
+        {
+            throw new AttributeAlreadyExistsException(string.Format("Method {0} already exists", name));
```

Comment:
We might have to do some revising of this to fit the string interpolation refactor I did yesterday